Bottom: d74b497b8946307890d9d77c390c156b49d5ca0f
Top:    19526f1c465544cbca3ac0c5f9639938f306860e
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-21 02:59:34 +0100

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index 92f9832..65a443b 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -309,6 +309,16 @@ original_poly (PolygonType * p)
       if (n == p->PointN - 1 ||
           (hole < p->HoleIndexN && n == p->HoleIndex[hole] - 1))
         {
+          if (contour == NULL)
+            {
+              printf ("How did that escape - did the loop iterate zero times??\n");
+              POLYGONPOINT_LOOP (p);
+                {
+                  printf ("Hello\n");
+                }
+              END_LOOP;
+              return NULL;
+            }
           poly_PreContour (contour, TRUE);
 
           /* make sure it is a positive contour (outer) or negative (hole) */
